back-end/routes/users.js

```diff
@@ -40,18 +40,25 @@ router.post('/login', [
     const query = `SELECT password, id FROM users WHERE email = '${email}'`;
     const dbResults = await getRowFromDb(query);
 
-    // value changes if query worked.
-    const passwordCheck = dbResults.password;
-    const id = dbResults.id;
 
-    // check if user-supplied password matches the hashed password from the db
-    if(await bcrypt.compare(password, passwordCheck)){
-      // if it does, generate and return a jwt
-      const token = createJWT(id);
+    // update: need to check if dbResults returned undefined (i.e. does user exist?)
+    if(typeof dbResults !== 'undefined') {
+      // value changes if query worked.
+      const passwordCheck = dbResults.password;
+      const id = dbResults.id;
 
-      res.status(200).send({ token, id });
+      // check if user-supplied password matches the hashed password from the db
+      if (await bcrypt.compare(password, passwordCheck)) {
+        // if it does, generate and return a jwt
+        const token = createJWT(id);
+
+        res.status(200).send({token, id});
+      } else {
+        res.status(403).send('Login attempt failed.');
```

Comment:
These two 403s can probably just be renamed to 'Invalid username or password'. I do not think it's recommended to let the user know that a user exists in the db. Not a big deal for this project though.